### DISCUSSION TITLE
Replicate the signac gene activity score model

### DISCUSSION THREAD

**Discussion by myylee** [GreenleafLab/ArchR#1033] [Questions / Documentation] [answered]
Hi, thanks for the great package. 

I want to implement the same gene activity score model used in the Signac package, which is the sum of reads mapped to gene body + 2kb upstream. 

I'm not sure if the parameters I have for the addGeneScoreMatrix are correct, anyone suggestion/comment will help! 

proj <- addGeneScoreMatrix(proj,geneModel="1",
                           matrixName="GeneScoreMatrix_signac",
                           geneUpstream = 2000,
                           logFile = createLogFile("addGeneScoreMatrix_signac"))

**Comment by rcorces** [comment, marked as answer]
If you are looking to exactly replicate Signac, I dont think I can help you with that since I do not know anything more about Signac's current gene score model than what you put in your question. i.e. I dont know if Signac uses gene boundaries, scale factors, etc. Here is my best attempt:
```
addGeneScoreMatrix(
  input = proj,
  geneModel = "1",
  matrixName = "GeneScoreMatrix_signac",
  extendUpstream = c(0, 0),
  extendDownstream = c(0, 0),
  geneUpstream = 2000,
  geneDownstream = 0,
  useGeneBoundaries = FALSE,
  useTSS = FALSE,
  tileSize = 500,
  ceiling = 4,
  geneScaleFactor = 5,
  scaleTo = 10000,
  excludeChr = c("chrY", "chrM"),
  blacklist = getBlacklist(input),
  threads = getArchRThreads(),
  parallelParam = NULL,
  subThreading = TRUE,
  force = FALSE,
  logFile = createLogFile("addGeneScoreMatrix")
)
```

**Reply by chen-zhan** [reply]
Hi, thanks for your kindness!

 I meet an error "<simpleError in .logError(e, fn = ".addGeneScoreMat TmpGS", info = sampleName,     errorList = errorList, logFile = logFile): Exiting See Error Above>".  So I try to run `proj = addGeneScoreMatrix(proj,c=FALSE)`.

And I read introduction of "addGeneScoreMatrix". But I am double about the result of switch addGeneScoreMatrix from TRUE to FALSE.
It will influent my data badly?
